wikimedia/wikipedia-ios
# Update to Xcode 13.1

**Phabricator:** 
https://phabricator.wikimedia.org/T291382

### Notes
This attempts to fix bugs that building with Xcode 13.1 surfaces, as well as update various settings files to the latest (Circle CI tests, build server version target, etc.)

### Test Steps
These are the areas that changed, so here's some optional regression test steps you can go through in iOS 13, 14, & 15. I will also add these to the Phabricator task for QA to do heavier testing with.
1. Navigation bars. Particularly Settings navigation bars and the Editing view navigation bar, where I was seeing UI bugs. Sometimes they showed as clear for me, with the UI underneath poking through. Test theming for these as well.
2. The tab bar. Similar to navigation bars, it's background color sometimes showed as clear. Test changing the theme through Settings, Article view, and the Editing flow, and be sure the tab bar updates correctly.
3. Settings section headers and footers theming and spacing. These weren't theming well on iOS15. I updated these to a custom header / footer view (which we were already halfway using on these screens) which already themed properly.
4. Collection view cell & header/footer sizes - made a change to fix a console error here. Look at Explore feed, Saved article tab, History tab, and Search results cell sizing.
5. Announcements deserialization - launch app, background, foreground, then pull to refresh on Explore feed. Announcements will be fetched. Before https://github.com/wikimedia/wikipedia-ios/pull/4075/commits/c7e762f89395bbf9e7d3abb33d9c074918a63c06, a console error would show at this point with something like:

```
 -[NSKeyedUnarchiver validateAllowedClass:forKey:] allowed unarchiving safe plist type 'NSString' for key 'NS.objects', even though it was not explicitly included in the client allowed classes set: '{(
    "'WMFMTLModel',
    "'NSArray'
)}'. This will be disallowed in the future.
```

Confirm this console error no longer shows. 

To confirm fundraising announcement still works:
1. Go to an article, confirm no fundraising announcement shows. 
2. Enter airplane mode and terminate the app.
3. Change device date time to sometime in December.
4. Relaunch app via Xcode. Confirm you still see no NSKeyedUnarchiver console errors.
5. Go to an article. Confirm you see fundraising announcement. Dismiss popup.
6. Revisit article, confirm you no longer see fundraising announcement. Terminate app.
7. Relaunch app, visit article, confirm you still no longer see fundraising announcement.
